pandas/io/parsers.py

```diff
@@ -1786,12 +1786,7 @@ def _handle_usecols(self, columns, usecols_key):
                 if len(columns) > 1:
                     raise ValueError("If using multiple headers, usecols must "
                                      "be integers.")
-                col_indices = []
-                for u in self.usecols:
-                    if isinstance(u, string_types):
-                        col_indices.append(usecols_key.index(u))
-                    else:
-                        col_indices.append(u)
+                col_indices = Index(usecols_key).get_indexer_for(self.usecols)
```

Comment:
```
In [4]: Index(list('aab')).get_indexer_for(list('ab'))
Out[4]: Int64Index([0, 1, 2], dtype='int64')
```

``Index(usecols_key).get_indexer_for(pd.unique(self.usecols))`` might do the trick